src/main/java/baritone/map/MapChunk.java

```diff
@@ -0,0 +1,98 @@
+import baritone.bot.utils.BlockStateInterface;
+import baritone.bot.utils.pathing.BetterBlockPos;
+import net.minecraft.block.material.MapColor;
+import net.minecraft.init.Blocks;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.BlockPos.MutableBlockPos;
+import net.minecraft.world.chunk.Chunk;
+import org.lwjgl.util.Color;
+
+import javax.imageio.ImageIO;
+import java.awt.*;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+public class MapChunk {
+
+    private Chunk chunk;
+
+    public MapChunk(Chunk chunk) {
+        this.chunk = chunk;
+    }
+
+    public void writeImage() {
+        Path image = getImagePath();
+        if(!Files.exists(image.getParent())) {
+            try {
+                Files.createDirectory(image.getParent());
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+        BufferedImage bufferedImage = new BufferedImage(16, 16, BufferedImage.TYPE_INT_RGB);
+        for(int x = 0; x < 16; x++) {
+            for(int z = 0; z < 16; z++) {
+                bufferedImage.setRGB(x, z, getColor(x, z));
+            }
+        }
+        try {
+            ImageIO.write(bufferedImage, "PNG", image.toFile());
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    public Path getImagePath() {
+        return new File(new File(Baritone.INSTANCE.getDir(), "map"), "chunk" + chunk.x + "-" + chunk.z + ".png").toPath();
+    }
+
+    /**
+     * getColor is a re-implementation of the Minecraft ItemMap's
+     * surface mapping system. This is significantly less convoluted
+     * than the poorly de-obfuscated code.
+     *
+     * We perform shading of the given coordinate relative to the
+     * position one to the north of the coordinate, as can be seen
+     * in Minecraft's maps.
+     *
+     * @param x relative x coordinate to the start of the chunk
+     * @param z relative z coord.
+     * @return Integer RGB value with contextual shading
+     */
+    public int getColor(int x, int z) {
+        int chunkX = chunk.getPos().getXStart() + x;
+        int chunkZ = chunk.getPos().getZStart() + z;
+        MutableBlockPos blockPos = new MutableBlockPos(chunkX, chunk.getHeight(new BetterBlockPos(chunkX, 0, chunkZ)), chunkZ);
+        MapColor color = BlockStateInterface.get(blockPos).getMapColor(chunk.getWorld(), blockPos);
+
+        // The chunk heightMap returns the first non-full block above the surface, including bushes.
+        // We have to check this and shift our block down by one if the color is "air" (as in, there's no real block there)
+        int colorValue = color != MapColor.AIR ? color.colorValue : BlockStateInterface.get(blockPos.down()).getMapColor(chunk.getWorld(), blockPos).colorValue;
+        int red = (colorValue >> 16) & 0xFF;
```

Comment:
Idk, I'm a big fan of something like:

```
Color color = new Color(colorValue);
int red = color.getRed();
int green = color.getGreen();
int blue = color.getBlue();
```

just for clarity on what's going on, bitshifts and &s are a little unclear

Reply:
This, in the context of https://github.com/cabaletta/baritone/pull/66#discussion_r211692811 is a good change.